pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -477,93 +546,214 @@ func (a *ApplicationRestoreController) restoreVolumes(restore *storkapi.Applicat
+						resourceExport.Namespace = restore.Namespace
+						resourceExport.Spec.Type = kdmpapi.ResourceExportBackup
+						// TODO: In the restore path we need to change source and destination ref as it is confusing now
+						// Usually dest means where it's backed up or restore to
+						source := &kdmpapi.ResourceExportObjectReference{
+							APIVersion: restore.APIVersion,
+							Kind:       restore.Kind,
+							Namespace:  restore.Namespace,
+							Name:       restore.Name,
+						}
+						backupLocation, err := storkops.Instance().GetBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
+						if err != nil {
+							return fmt.Errorf("error getting backup location path: %v", err)
+						}
+						destination := &kdmpapi.ResourceExportObjectReference{
+							// TODO: GetBackupLocation is not returning APIVersion and kind.
+							// Hardcoding for now.
+							// APIVersion: backupLocation.APIVersion,
+							// Kind:       backupLocation.Kind,
+							APIVersion: "stork.libopenstorage.org/v1alpha1",
+							Kind:       "BackupLocation",
+							Namespace:  backupLocation.Namespace,
+							Name:       backupLocation.Name,
+						}
+						resourceExport.Spec.TriggeredFrom = kdmputils.TriggeredFromStork
+						storkPodNs, err := k8sutils.GetStorkPodNamespace()
+						if err != nil {
+							logrus.Errorf("error in getting stork pod namespace: %v", err)
+							return err
+						}
+						resourceExport.Spec.TriggeredFromNs = storkPodNs
+						resourceExport.Spec.Source = *source
+						resourceExport.Spec.Destination = *destination
+						_, err = kdmpShedOps.Instance().CreateResourceExport(resourceExport)
+						if err != nil {
+							logrus.Errorf("failed to create resourceExport CR %v: %v", crName, err)
+							return err
+						}
+						return nil
+					}
+					logrus.Errorf("%s error reading resourceExport CR %v: %v", funct, crName, err)
+					return nil
+				} else {
+					var message string
+					logrus.Infof("%s re cr %v status %v", funct, crName, resourceExport.Status.Status)
+					switch resourceExport.Status.Status {
+					case kdmpapi.ResourceExportStatusFailed:
+						message = fmt.Sprintf("%s Error creating CR %v for pvc creation: %v", funct, crName, err)
+						restore.Status.Status = storkapi.ApplicationRestoreStatusFailed
+						restore.Status.Stage = storkapi.ApplicationRestoreStageFinal
+						restore.Status.Reason = message
+						restore.Status.LastUpdateTimestamp = metav1.Now()
+						err = a.client.Update(context.TODO(), restore)
+						if err != nil {
+							return err
+						}
+						a.recorder.Event(restore,
+							v1.EventTypeWarning,
+							string(storkapi.ApplicationRestoreStatusFailed),
+							message)
+						log.ApplicationRestoreLog(restore).Errorf(message)
+						return err
+					case kdmpapi.ResourceExportStatusInitial:
+						return nil
+					case kdmpapi.ResourceExportStatusPending:
+						return nil
+					case kdmpapi.ResourceExportStatusInProgress:
+						return nil
```

Comment:
We need to update LastUpdateTimestamp in initial, pending and in-progress states to avoid timeout